Add cooldown placeholder + QOL changes

Janmm14 opened the pull request:
- Add cooldown placeholder with arbitrarykey
 - Add /cc cooldown <arbitrarykey> command to mark cooldown used
 - Add /cc reload command
 - When command execution delay is 0 or less, execute immediately. Delay 0 in scheduler results to next tick execution
 - .gitignore *.iml files

I am not happy with the wording in the Readme though

Janmm14 committed:
Add cooldown placeholder + QOL changes

 - Add cooldown placeholder with arbitrarykey
 - Add /cc cooldown <arbitrarykey> command to mark cooldown used
 - Add /cc reload command
 - When command execution delay is 0 or less, execute immediately. Delay 0 in scheduler results to next tick execution
 - .gitignore *.iml files

konsolas commented on README.md:
````diff
@@ -41,8 +41,13 @@ Placeholders are delimited by '-'. Since they're applied with a replace, errors
+ - ```aacvl:<check>``` - AAC 1-4 violation level of the given check (internal name). e.g. ```-aacvl:speed-```
  - ```chance:<percentage>%``` - Will be 1.0 percentage% of the time. e.g. ```-chance:34.5%-```
+ - ```cooldown:<arbitrarykey>#<seconds>``` - Returns 0 if ```<arbitrarykey>``` was not seen before by ```/cc cooldown <arbitrarykey>``` command.
+   Otherwise returns time in seconds until last execution of the cooldown command for the given ```<arbitrarykey>``` is more than ```<seconds>``` ago. Never returns a value < 0.
````
Maybe:

Returns `0` if `/cc cooldown` has never been executed for the given arbitrarykey, or if it was last executed more than `<seconds>` ago. Otherwise, returns time in seconds since the last execution of the cooldown command.

Janmm14 committed:
Improve readme for cooldown placeholder

Co-authored-by: Vincent <konsolas@users.noreply.github.com>

Janmm14 commented:
I just noticed that the descirption you suggested is wrong. I return seconds until cooldown is over.

I will change it to be easier in usage and in code, it will then jsut return seconds since /cc cooldown arbitrarykey was last executed.
If it was not executed yet, I will return 43200

Janmm14 committed:
Simplify cooldown system

konsolas commented on src/main/java/me/konsolas/conditionalcommands/placeholders/PlaceholderCooldown.java:
```diff
@@ -0,0 +1,38 @@
+            if (diff < 1) {
+                return "0";
+            } else {
+                return Long.toString(diff);
```
Could this return a value > 43200? It seems that Guava caches don't always perform evictions on read, so maybe it's worth putting a `Math.min(43200, diff)` here. See https://github.com/google/guava/wiki/CachesExplained#when-does-cleanup-happen

konsolas commented on README.md:
````diff
@@ -42,8 +42,12 @@ Placeholders are delimited by '-'. Since they're applied with a replace, errors
+ - ```aacvl:<check>``` - AAC 1-4 violation level of the given check (internal name). e.g. ```-aacvl:speed-```
  - ```chance:<percentage>%``` - Will be 1.0 percentage% of the time. e.g. ```-chance:34.5%-```
+ - ```cooldown:<arbitrarykey>``` - Returns time in seconds since the last execution of the ```/cc cooldown <arbitrarykey>``` command. Returns ```43200``` if ```/cc cooldown``` has never been executed for the given ```<arbitrarykey>```. Never returns a value < 0.
+   e. g. ```-cooldown:dont_spam_me-```
````
Maybe add a more concrete example like

```
/ccmd if -cooldown:some_key->60 do /0/ ccmd cooldown some_key /0/ broadcast sent up to once every 60 seconds
```

if the above works

Janmm14 committed:
Apply requested changes

- Complete example for cooldown in readme
- Ensure we actually never return a value > 43200, as the cache does not strictly evict.

konsolas commented on src/main/java/me/konsolas/conditionalcommands/placeholders/PlaceholderCooldown.java:
```diff
@@ -0,0 +1,38 @@
+            if (diff < 1) {
+                return "0";
+            } else {
+                return Long.toString(Math.min(diff, 3200));
```
Meant to be 43200?

Janmm14 replied:
rip typo
```suggestion
                return Long.toString(Math.min(diff, 43200));
```

Janmm14 committed:
Fix typo
